debian/patches/0004-add-cuda-tonemap-impl.patch

```diff
@@ -2396,6 +2412,25 @@ Index: FFmpeg/libavfilter/vf_tonemap_cuda.c
 +    if ((ret = get_rgb2rgb_matrix(in_pri, out_pri, rgb2rgb_matrix)) < 0)
 +        return ret;
 +
++    if (s->in_desc->comp[0].depth == 10) {
++        input_quantization_offset = 0.0009765774014f;
++        input_y_scale = 1.1678082192f;
++        input_uv_scale = 1.1417410714f;
++    } else if (s->in_desc->comp[0].depth == 16) {
++        input_quantization_offset = 0.0002441443503f;
++        input_y_scale = 1.1689497717f;
++        input_uv_scale = 1.1428571429f;
++    }
++
++    if (s->out_desc->comp[0].depth == 10) {
++        output_quantization_offset = 0.0009765774014f;
++    }
++
++    if (s->out_desc->comp[0].depth > 8) {
++        output_quantization_factor = 256.0f;
++        output_quantization_scale = 65535.0f;
++    }
```

Comment:
For hard-coded values ​​that are not obvious, how about using formulas in the host whenever possible and adding some comments in the kernel?

Reply:
These values' derivation process is also not obvious, but all of them are derived from BT.2100 TABLE 9's Quantization Formula.

Reply:
Something like this?
```c
    if (ctx->in_desc->comp[0].depth >= 10 &&
        ctx->in_desc->comp[0].depth <= 16) { // not sure how to handle P016 (12bit)
        int idepth = ctx->in_desc->comp[0].depth;
        av_bprintf(&header, "#define INPUT_QUANTIZATION_OFFSET %ff\n",
                   ((256 >> (idepth - 8)) / 65535.0f));
        av_bprintf(&header, "#define INPUT_Y_SCALE %ff\n",
                   ((float)((256 << (idepth - 8)) - 1) / (219 << (idepth - 8)));
        av_bprintf(&header, "#define INPUT_UV_SCALE %ff\n",
                   ((float)((256 << (idepth - 8)) - 1) / (224 << (idepth - 8)));
    } else {
        // should not happen, but for completeness
        av_bprintf(&header, "#define INPUT_QUANTIZATION_OFFSET 0.0f\n");
        av_bprintf(&header, "#define INPUT_Y_SCALE 1.0f\n");
        av_bprintf(&header, "#define INPUT_UV_SCALE 1.0f\n");
    }

    if (ctx->out_desc->comp[0].depth >= 10) {
        av_bprintf(&header, "#define RESCALE_LIMITED_RANGE_OUTPUT\n");
    }

    if (ctx->out_desc->comp[0].depth >= 10 &&
        ctx->out_desc->comp[0].depth <= 16) { // not sure how to handle P016 (12bit)
        int odepth = ctx->out_desc->comp[0].depth;
        av_bprintf(&header, "#define OUTPUT_QUANTIZATION_OFFSET %ff\n",
                   ((256 >> (odepth - 8)) / 65535.0f));
    }

...

    y = INPUT_Y_SCALE * y - 0.07305936073f; // -(16.0f  / 219.0f)
    u = INPUT_UV_SCALE * u - 0.5714285714f; // -(128.0f / 224.0f)
    v = INPUT_UV_SCALE * v - 0.5714285714f;
```

Reply:
That is still not quite enough. I define all constants in `colorspace.h` and write down the derivation process there